sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInAsyncShouldBeWrapped.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Tests.Diagnostics
+{
+    public class InvalidCases
+    {
+        public static async Task SkipLinesAsync(this TextReader reader, int linesToSkip) // Noncompliant {{Split this method into two, one handling parameters check and the other handling the iterator.}}
```

Comment:
It would be useful to point to where the exceptions are thrown (in secondary locations).